METALIUM_GUIDE.md

```diff
@@ -114,6 +114,152 @@ TODO: Describe that TT wins at scale-out, best compute density at the server an
   <img width="1176" alt="image" src="https://github.com/tenstorrent/tt-metal/assets/3885633/d3c89155-6e4d-49cb-a95c-85654ac29e7d">
 <img width="1171" alt="image" src="https://github.com/tenstorrent/tt-metal/assets/3885633/73039d17-3bce-4ff5-b797-da1aa9b147c4">
 
+#### Behind the scenes, a Compute Kernel becomes Unpack, Math, Pack Kernels
+
+As the above single Tensix core architecture shows, the Compute Kernel uses
+three Baby RISCVs. tt-metal uses them to conduct unpack, math, and pack. It
+actually generates Unpack, Math, Pack kernels from a single Compute kernel.
+
+For details, let's check what tt-metal generates for the following compute
+kernel:
+```
+namespace NAMESPACE {
+void MAIN {
+  mm_init();
+  acquire_dst(tt::DstMode::Tile);
```

Comment:
this is great, please also use `tile_regs_acquire` , `tile_regs_commit` , `tile_regs_wait` , `tile_regs_release`,   , the new  math->pack sync APIs,

these will further clarify the double buffer hand-shake on the tile_regs registers

Reply:
Ah, I did not know them. Updated. Thanks!